src/main/java/org/folio/circulation/domain/UserBlRepository.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.circulation.domain;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.FetchSingleRecord;
+import org.folio.circulation.support.Result;
+
+import io.vertx.core.json.JsonObject;
+
+public class UserBlRepository {
```

Comment:
Why is this separate to the user repository?

I think it would be better to hide away that this has to use different storage APIs from the rest of the system. This naming leaks the storage implementation details as opposed to the purpose which is about users.

If the context is more specific than users, please rename this class to something more specific like `LoggedInUserFinder`.

Reply:
This class is a connector to the functionality in the whole another module: mod-bl-users. We do use one method now.

Reply:
I think the class should be named after what it's purpose or role is, not necessarily how it achieves it.

(though I'd prefer it be both in the case of repositories e.g. `StorageUserRepository`)

Reply:
Agree with you regarding class naming. In this case, the code has been refactored to not use mod-bl-users at all.